### ISSUE TITLE
Backend needs Github Actions to run tests for it

### ISSUE THREAD

**Issue by FernFinder** [UNLV-CS472-672/2024-S-GROUP7-LifeQuest#81]
## Describe the issue

The backend for our web app is currently not being tested using GitHub Actions. We will need to implement a change to our current workflow to fix this.

## To Reproduce

See current workflow and how it reviews only our frontend.

## Expected behavior

Backend should be tested.

**Comment by dusek2** [comment]
We will have to use github secrets to add mongoDB URI into the environment
Also we will need to run Front End, Backend and the unit tests simultaneously

### Here is advice from ChatGPT4:
https://chat.openai.com/share/fd583dea-c5d5-4513-813e-e543156dc5bc

To set up simultaneous running applications in GitHub Actions and securely use secrets, such as a MongoDB URI, follow these summarized steps:

- **Add Secrets to GitHub Repository:**
  - Navigate to your repository's Settings > Secrets > Actions.
  - Click on "New repository secret" to add each secret, like your MongoDB URI, naming it appropriately (e.g., `MONGO_DB_URI`).

- **Configure GitHub Actions Workflow:**
  - **Checkout Code:** Use `actions/checkout@v2` to check out your repository code.
  - **Setup Node.js:** Utilize `actions/setup-node@v2` for setting up Node.js environment.
  - **Install Dependencies:** Run `npm install` commands for both frontend and backend directories if applicable.
  - **Use Secrets:** Inject secrets into your workflow using the `env` or `$GITHUB_ENV` method to set environment variables. E.g., `echo "MONGODB_URI=${{ secrets.MONGO_DB_URI }}" >> $GITHUB_ENV`.
  - **Start Backend and Frontend:** Execute commands to start your backend and frontend applications in the background. Use `&` to run processes in the background and `sleep` to allow time for services to start up.
  - **Run Tests:** Finally, run your unit tests or other tests that might require interaction with the started services.

- **Environment Variables in Application:**
  - Access the MongoDB URI (or other secrets) in your application code through `process.env.MONGODB_URI` or the relevant environment variable.

### Example Workflow Step for Starting Services and Running Tests:

```yaml
    - name: Start Backend
      run: cd backend && npm start &
    - name: Start Frontend
      run: cd frontend && npm start &
    - name: Wait for Backend and Frontend to be ready
      run: sleep 30
    - name: Run Tests
      env:
        MONGODB_URI: ${{ secrets.MONGO_DB_URI }}
      run: cd backend && npm test
```

### Key Points:

- **Security:** Always use GitHub Secrets for sensitive information to avoid exposing them in your code or workflow files.
- **Parallel Execution:** Use background processes for simultaneous service execution and a delay (`sleep`) to ensure services are fully up before proceeding.
- **Environment Variables:** Pass secrets as environment variables to your application, ensuring secure and flexible configuration management.

This setup allows you to run both your frontend and backend environments alongside each other in a CI environment, enabling comprehensive integration and end-to-end tests without exposing sensitive configuration details.

**Comment by FernFinder** [comment]
I think I should be able to get this running tomorrow 4/15.

**Comment by thegreatzoidberg** [comment]
Fixed by [Pull #99](https://github.com/UNLV-CS472-672/2024-S-GROUP7-LifeQuest/pull/99).
